ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -0,0 +1,244 @@
+
+    final List<String> columns = insertValues.getColumns().isEmpty()
+        ? dataSource.getSchema()
+          .fields()
+          .stream()
+          .map(Field::name)
+          .filter(name -> !SchemaUtil.ROWTIME_NAME.equals(name))
+          .collect(Collectors.toList())
+        : insertValues.getColumns();
+
+    if (columns.size() != insertValues.getValues().size()) {
+      // this will only happen if insertValues.getColumns() is empty, otherwise
+      // the columns/values are verified by the InsertValues constructor
+      throw new KsqlException(
+          "Expected a value for each column. Expected Columns: " + columns
+              + ". Got " + insertValues.getValues());
+    }
+
+    final Map<String, Object> values = new HashMap<>();
+    for (int i = 0; i < columns.size(); i++) {
+      final String column = columns.get(i);
+      final Schema columnSchema = dataSource.getSchema().field(column).schema();
+      final Expression valueExp = insertValues.getValues().get(i);
+
+      values.put(column, new ExpressionResolver(columnSchema, column).process(valueExp, null));
+    }
+
+    if (keyField.isPresent()) {
+      final String key = keyField.get();
+      final Object keyValue = values.get(key);
+      final Object rowKeyValue = values.get(SchemaUtil.ROWKEY_NAME);
+
+      if ((keyValue != null) ^ (rowKeyValue != null)) {
+        values.putIfAbsent(key, rowKeyValue);
+        values.putIfAbsent(SchemaUtil.ROWKEY_NAME, keyValue);
+      } else if (!Objects.equals(keyValue, rowKeyValue)) {
+        throw new KsqlException(
+            String.format(
+                "Expected ROWKEY and %s to match but got %s and %s respectively.",
+                key, rowKeyValue, keyValue));
+      }
+    }
+
+    values.putIfAbsent(SchemaUtil.ROWTIME_NAME, clock.getAsLong());
+
+    return new GenericRow(
+        dataSource.getSchema()
+            .fields()
+            .stream()
+            .map(Field::name)
+            .map(values::get)
+            .collect(Collectors.toList())
+    );
+  }
+
+  @SuppressWarnings("unchecked") // we know that key is String
+  private byte[] serializeKey(final Object keyValue, final StructuredDataSource<?> dataSource) {
+    if (keyValue == null) {
+      return null;
+    }
+
+    return ((Serde<String>) dataSource.getKeySerdeFactory().create())
+        .serializer()
+        .serialize(dataSource.getKafkaTopicName(), keyValue.toString());
+  }
+
+  private byte[] serializeRow(
+      final GenericRow row,
+      final StructuredDataSource<?> dataSource,
+      final KsqlConfig config,
+      final ServiceContext serviceContext
+  ) {
+    final Serde<GenericRow> rowSerde = dataSource.getKsqlTopicSerde()
+        .getGenericRowSerde(
+            dataSource.getSchema(),
+            config,
+            serviceContext.getSchemaRegistryClientFactory(),
+            "insertValues",
+            ProcessingLogContext.create());
```

Comment:
It doesn't make sense to pass in a processing log context here. For insert into the serde should just throw whatever exception it hits (which should never happen) and the error should get sent back in the response as a 500. So we should change the serdes to make supplying a processing logger optional.

Reply:
Instead of littering optionals everywhere, I'll create a NO_OP processing logger.